ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt

```diff
@@ -169,4 +175,21 @@ internal constructor(
   }
 
   fun getZoomThresholdCrossed(): Observable<Nil> = zoomThresholdCrossed
+
+  private fun getLocationOfInterestFeatures(survey: Survey): Flow<Set<Feature>> =
+    loiRepository.getLocationsOfInterest(survey).map {
+      it.map { loi -> loi.toFeature() }.toPersistentSet()
+    }
```

Comment:
nit: Can we get rid of the nesting by using `flatMap`?

Reply:
`flatMap` pulls the elements of a list or flow up a level. That would change the output to `Flow<Feature>`, which is not what we want here; we want a full snapshot of all the displayed features on each emission.